File: 2015/1/Lisp.java
import java.util.Scanner;

public class Lisp{
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        System.out.printf("\nPlease enter your pattern: ");
        String inputString = input.next();
        input.close();

        int floor = 0;
        for(int i = 0; i < inputString.length(); i++){
            if(inputString.charAt(i) == '('){
                floor++;
            } else if(inputString.charAt(i) == ')'){
                floor--;
            } else{
                System.err.println("There where invalid characters in the pattern");
                return;
            }
        }

        System.out.printf("Floor %d\n", floor);
    }
}

File: 2015/1/lisp.cpp
#include <iostream>
#include <vector>
#include <string>

int main(void){
    std::string inputString;

    std::cout << "Please enter your Pattern";
    std::cin >> inputString;

    int floor;
    for(int i = 0; i < inputString.length(); i++){
        if(inputString.at(i) == '('){
            floor++;
        } else if(inputString.at(i) == ')'){
            floor--;
        } else{
            std::cout << "Error in the Pattern" << std::endl;
            return -1;
        }
        std::cout << inputString;
    }
    std::cout << "floor " << floor << std::endl;
}